internal/resolution/resolver_test.go

```diff
@@ -119,10 +123,11 @@ var _ = Describe("OperatorResolver", func() {
 	It("should return an error if the client throws an error", func() {
 		client := NewFailClientWithError(fmt.Errorf("something bad happened"))
 		entitySource := input.NewCacheQuerier(testEntityCache)
-		resolver := resolution.NewOperatorResolver(client, entitySource)
+		variableSource := olm.NewOLMVariableSource(client)
+		resolver := resolution.NewOperatorResolver(entitySource, variableSource)
 		solution, err := resolver.Resolve(context.Background())
 		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
+		Expect(err).To(Equal(fmt.Errorf("something bad happened")))
```

Comment:
Not sure what the common pattern is in this repo, but if the error is not an exported type, it should be treated as opaque and changing the literal text inside should not fail tests in my opinion - the presence of the error is the contract, not a particular sentinel value

Reply:
I think in this case it makes sense to check specific error because this test is about making sure that we return an error from the client (not from other parts of the resolution).

But this is out of scope for this PR. I'll revert in the next update.

Reply:
If that's important, I would export an error type and use `errors.Is()` to check it - prefer the type system over stringly typing.

Reply:
But do we want this specific error to be a user-facing API? I suppose we could put the error type in an internal package somewhere that's available to the test. And then all third party clients still see this error opaquely.

Reply:
But, my two cents, that seems like a lot of effort and moving parts to avoid string comparisons of error messages in tests.

Reply:
Wanting to inspect the black box to figure out which error it is without letting users do it is weird. If there's a real need to differentiate between these errors from this call-site, doing it via stringly-typed patterns is not a good idea.